topic0/src/topic0_KeyPoint2/BuildingDirector.java

```diff
@@ -0,0 +1,22 @@
+package topic0_KeyPoint2;
+
+public class BuildingDirector {
+	
+	private BuildingBuilder buildingBuilder = null;
```

Comment:
not necesarry to assign to null, if the constructor will be initializing this field.